### PR TITLE
Populating the interactive tutorials.

### PR DESCRIPTION
**Pull request by BradLarson**
This fills out the initial interactive tutorials.

The following tutorials have been completed:

- [X] Differentiable functions 
- [X] Differentiable types
- [x] Gradient descent

### CODE REVIEW

File: Sources/DifferentiableSwiftExamplesDocumentation/DifferentiableSwiftExamples.docc/Resources/Code/DifferentiableFunctions/DifferentiableFunctions-01-04.swift
Comment (JaapWijnen) [inline Sources/DifferentiableSwiftExamplesDocumentation/DifferentiableSwiftExamples.docc/Resources/Code/DifferentiableFunctions/DifferentiableFunctions-01-04.swift:1]: I'd prefer keeping the previous code as well. here the function definition is missing (same for the next step 01-05)

File: Sources/DifferentiableSwiftExamplesDocumentation/DifferentiableSwiftExamples.docc/Resources/Code/DifferentiableFunctions/DifferentiableFunctions-02-05.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+@derivative(of: squared)
+func vjpSquared(_ input: Double) -> (
+    value: Double,
+    pullback: (Double) -> Double
+) {
+    let output = squared(value)
+    func pullback(_ tangentVector: Double) -> Double {
+        return tangentVector * 2 * output)
```
Comment (JaapWijnen) [inline Sources/DifferentiableSwiftExamplesDocumentation/DifferentiableSwiftExamples.docc/Resources/Code/DifferentiableFunctions/DifferentiableFunctions-02-05.swift]: Parenthesis at the end should be removed.

Comment (JaapWijnen) [reply Sources/DifferentiableSwiftExamplesDocumentation/DifferentiableSwiftExamples.docc/Resources/Code/DifferentiableFunctions/DifferentiableFunctions-02-05.swift]: Also is this pullback correct? This looks like 2 x^2 dx

File: Sources/DifferentiableSwiftExamplesDocumentation/DifferentiableSwiftExamples.docc/Tutorials/GradientDescent.tutorial [hunk 2]
```diff
@@ -1,9 +1,75 @@
...
+            @Step {
+                Let's create the perceptron as a custom data type. The weights for the connections between neurons will be randomly-initialized floating point values, and the bias will start at zero.
+                @Code(name: "main.swift", file: "GradientDescent-01-02.swift")
+            }
+            @Step {
+                Because all of the properties of our perceptron conform to `Differentiable`, it's easy to make the perceptron itself differentiable by adding the conformance to `Differentiable`.
+                @Code(name: "main.swift", file: "GradientDescent-01-03.swift")
+            }
+            @Step {
+                A convenient language feature that was introduced to Swift is the ability of structs to be called as if they were functions. To do so, you need to define a `callAsFunction()` method on the struct.
```
Comment (JaapWijnen) [inline Sources/DifferentiableSwiftExamplesDocumentation/DifferentiableSwiftExamples.docc/Tutorials/GradientDescent.tutorial:30]: I feel adding the callAsFunction here is adding unnecessary complexity. I'm guessing this is inspired from a more pythony way of training a model/inferencing? I'd rather have a named function to avoid the added callAsFunction complexity.

**Review by JaapWijnen — CHANGES_REQUESTED** [review]

File: Sources/DifferentiableSwiftExamplesDocumentation/DifferentiableSwiftExamples.docc/Resources/Code/DifferentiableFunctions/DifferentiableFunctions-01-04.swift
Comment (BradLarson) [reply Sources/DifferentiableSwiftExamplesDocumentation/DifferentiableSwiftExamples.docc/Resources/Code/DifferentiableFunctions/DifferentiableFunctions-01-04.swift:1]: Thanks, fixed.

File: Sources/DifferentiableSwiftExamplesDocumentation/DifferentiableSwiftExamples.docc/Resources/Code/DifferentiableFunctions/DifferentiableFunctions-02-05.swift (hunk 1 above)
Comment (BradLarson) [reply Sources/DifferentiableSwiftExamplesDocumentation/DifferentiableSwiftExamples.docc/Resources/Code/DifferentiableFunctions/DifferentiableFunctions-02-05.swift]: You're right, this was incorrect. Both the parenthesis and the wrong derivative should be fixed now.

File: Sources/DifferentiableSwiftExamplesDocumentation/DifferentiableSwiftExamples.docc/Tutorials/GradientDescent.tutorial (hunk 2 above)
Comment (BradLarson) [reply Sources/DifferentiableSwiftExamplesDocumentation/DifferentiableSwiftExamples.docc/Tutorials/GradientDescent.tutorial:30]: `callAsFunction()` has been very heavily used in existing Swift ML models, and was important enough to the Swift for TensorFlow team that it was the subject of [this accepted Evolution proposal](https://github.com/apple/swift-evolution/blob/main/proposals/0253-callable.md). Many differentiable Swift models will have this kind of calling convention, which makes them very easy to read and work with, so I thought it valuable to provide a demonstration of that here.

Comment (JaapWijnen) [reply Sources/DifferentiableSwiftExamplesDocumentation/DifferentiableSwiftExamples.docc/Tutorials/GradientDescent.tutorial:30]: Feels very Pythony to me which I understand but I don't necessarily see the value we add by doing this and/or promoting this as a good approach. But I can accept it!

**Review by JaapWijnen — APPROVED** [review]
